src/TruePBR/BSLightingShaderMaterialPBRLandscape.cpp

```diff
@@ -0,0 +1,138 @@
+#include "BSLightingShaderMaterialPBRLandscape.h"
+
+BSLightingShaderMaterialPBRLandscape::BSLightingShaderMaterialPBRLandscape()
+{
+	std::fill(isPbr.begin(), isPbr.end(), false);
+	std::fill(roughnessScales.begin(), roughnessScales.end(), 1.f);
+	std::fill(displacementScales.begin(), displacementScales.end(), 1.f);
+	std::fill(specularLevels.begin(), specularLevels.end(), 0.04f);
+}
+
+BSLightingShaderMaterialPBRLandscape::~BSLightingShaderMaterialPBRLandscape()
+{}
+
+BSLightingShaderMaterialPBRLandscape* BSLightingShaderMaterialPBRLandscape::Make()
+{
+	return new BSLightingShaderMaterialPBRLandscape;
+}
+
+RE::BSShaderMaterial* BSLightingShaderMaterialPBRLandscape::Create()
+{
+	return Make();
+}
+
+void BSLightingShaderMaterialPBRLandscape::CopyMembers(RE::BSShaderMaterial* that)
+{
+	BSLightingShaderMaterialBase::CopyMembers(that);
+
+	auto* pbrThat = static_cast<BSLightingShaderMaterialPBRLandscape*>(that);
+
+	pbrThat->numLandscapeTextures = numLandscapeTextures;
+
+	for (uint32_t textureIndex = 0; textureIndex < NumTiles; ++textureIndex) {
+		pbrThat->landscapeBaseColorTextures[textureIndex] = landscapeBaseColorTextures[textureIndex];
```

Comment:
`textureIndex` could be equal to `NumTiles-1` which is greater than the array size.

Reply:
Fixed

Reply:
You can resolve all my previous comments now